app/services/control_rate_service.rb

```diff
@@ -109,23 +109,12 @@ def bp_quarterly_query(period)
       .order("patient_id, bp_recorded_at DESC, bp_id")
   end
 
-  private
-
   def quarterly_report?
     @quarterly_report
   end
 
   def cache_key
-    "#{self.class}/#{region.model_name}/#{region.id}/#{periods_cache_key}"
-  end
-
-  def periods_cache_key
-    value = if periods.is_a?(Range)
-      "#{periods.begin.value}/#{periods.end.value}"
-    else
-      period.value
-    end
-    "#{periods.begin.type}_periods/#{value}"
+    "#{self.class}/#{region.model_name}/#{region.id}/#{report_range.begin.type}/#{Date.current}"
```

Comment:
Our cache key now only cares about the period type (:month or :quarter), and the date that the queries are run.